docs/core/extensions/caching.md

```diff
@@ -0,0 +1,179 @@
+
+## In-memory caching
+
+In this section, you'll learn about the [`Microsoft.Extensions.Caching.Memory`](/dotnet/api/microsoft.extensions.caching.memory) package. The current implementation of the <xref:Microsoft.Extensions.Caching.Memory.IMemoryCache> is a wrapper around the <xref:System.Collections.Concurrent.ConcurrentDictionary%602>, exposing a feature-rich API. Entries within the cache are represented by the <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry>, and can be any `object`. The in-memory cache solution is great for apps that run in a single server, where all the cached data rents memory in the app's process.
+
+> [!TIP]
+> For multi-server caching scenarios, consider the [Distributed caching](#distributed-caching) approach as an alternative to in-memory caching.
+
+### In-memory caching API
+
+The consumer of the cache has control over both sliding and absolute expirations:
+
+- <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry.AbsoluteExpiration?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry.AbsoluteExpirationRelativeToNow?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry.SlidingExpiration?displayProperty=nameWithType>
+
+Setting an expiration will cause entries in the cache to be *evicted* if they're not accessed within the expiration time allotment. Consumers have additional options for controlling cache entries, through the <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryOptions>. Each <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry> is paired with <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryOptions> which exposes expiration eviction functionality with <xref:Microsoft.Extensions.Primitives.IChangeToken>, priority settings with <xref:Microsoft.Extensions.Caching.Memory.CacheItemPriority>, and controlling the <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry.Size?displayProperty=nameWithType>. Consider the following extension methods:
+
+- <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryExtensions.AddExpirationToken%2A?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryExtensions.RegisterPostEvictionCallback%2A?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryExtensions.SetSize%2A?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryExtensions.SetPriority%2A?displayProperty=nameWithType>
+
+### In-memory cache example
+
+To use the default <xref:Microsoft.Extensions.Caching.Memory.IMemoryCache> implementation, call the <xref:Microsoft.Extensions.DependencyInjection.MemoryCacheServiceCollectionExtensions.AddMemoryCache%2A> extension method to register all the required services with DI. In the following code sample, the generic host is used to expose the <xref:Microsoft.Extensions.Hosting.HostBuilder.ConfigureServices%2A> functionality:
```

Comment:
What if I'm not doing DI? What if just want to new-up an implementation of IMemoryCache?

Reply:
You can instantiate a `MemoryCache` like this (given options):

```csharp
MemoryCache cache = new(Options.Create<MemoryCacheOptions>(new MemoryCacheOptions()));
```

See the following:

- [`MemoryCache.ctor`](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs#L36)
- [`Options.Create<TOptions>(TOptions)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.options.create?view=dotnet-plat-ext-5.0)

Reply:
Should that be mentioned?

Reply:
It is my understanding that since this is all based on the `Microsoft.Extensions.*` namespaces, and specific to fundamentals - we would prefer to have consumers not instantiate things outside of DI. While it is possible, it's best to use the infrastructure as a basis moving forward. It's like saying, we have this perfect three-layer chocolate cake -- but you can't eat it. 🎂